### PR TITLE
replace context with provider in tests

### PR DESCRIPTION
**Pull request by cforkish**
this is preliminary groundwork for upgrading react and redux. some of our tests were using a deprecated api to pass a redux store into a component, which has been removed in the latest version of react-redux. however, since fixing those tests to work with the latest version of react-redux doesn't require any functionality we don't already have in our existing version, we are safe to update these tests before updating react-redux.

while updating those tests i also updated them to use `reconfiguredChai`, and many of the code changes in this PR are related to that change

### CODE REVIEW

File: apps/test/unit/templates/CertificateTest.js [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+import {mount} from 'enzyme';
+import {expect} from '../../util/reconfiguredChai';
 import Certificate from '@cdo/apps/templates/Certificate';
 import {combineReducers, createStore} from 'redux';
 import responsive from '@cdo/apps/code-studio/responsiveRedux';
+import isRtl from '@cdo/apps/code-studio/isRtlRedux';
+
+const store = createStore(combineReducers({responsive, isRtl}));
+
+function wrapperWithTutorial(tutorial) {
```
Comment (maddiedierker) [inline apps/test/unit/templates/CertificateTest.js:12]: super nit: we could use a fat-arrow function here for consistency

Comment (maddiedierker) [reply apps/test/unit/templates/CertificateTest.js:12]: i won't make this comment everywhere for brevity, but if you make this update you'll have to make it in multiple places

Comment (cforkish) [reply apps/test/unit/templates/CertificateTest.js:12]: i started working on some [style guide update proposals](https://docs.google.com/document/d/1tZHMt6VHcAZzotdxjCHqZAbNn8dsnoj0QTiiabkLrEE/edit?usp=sharing) with Jessie a few months ago, but then got sucked into this huge progress table refactor project and haven't found time to return to it. but tl;dr, i think we should move away from arrow functions as a general practice, except for simple inline functions.

why? because it turns out using arrow functions as class properties can have [really bad performance implications](https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1). so since we want to discourage using them in that case, and since in the general case the only difference is one of style, i think it makes sense to be consistent and never use them (except when conforming to the style of an existing (non-component) module. not only would that make our style more consistent, but it would make it less likely that engineers who haven't memorized our style guide would accidentally use them as class properties in components.

anyway, until i get around to finishing up those proposals to present to the group, i've just been avoiding using them myself. currently our style guide is not opinionated about this and we have loads of examples of both patterns in our codebase.

tl;dr i would prefer to leave these as `function` functions.

Comment (maddiedierker) [reply apps/test/unit/templates/CertificateTest.js:12]: interesting, i didn't realize the performance implications of using arrow functions -- thank you for sharing that article! i'm fine with leaving these functions as-is, but agreed that we need to make our style guide more opinionated (which i'd be interested in partnering with you on!)

**Review by maddiedierker — APPROVED** [review]
